src/main/java/duke/commands/CreateDeadline.java

```diff
@@ -0,0 +1,48 @@
+package duke.commands;
+
+import java.time.LocalDateTime;
+
+import duke.ui.Ui;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.Deadline;
+import duke.storage.Storage;
+import duke.parsers.CommandParser;
+import duke.exceptions.DukeException;
+
+/**
+ * Creates a Deadline Task and adds it to the TaskList.
+ */
+class CreateDeadline extends Command {
+
+    public CreateDeadline(CommandParser commandParser) {
+        super(commandParser);
+    }
+
+    public void execute(String arg, TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        // Obtain command arguments from argument string
+        String[] args = split(arg, "/by");
+        String taskName = getTaskName(args);
+        LocalDateTime dateTime = getDateTime(args);
+
+        Task newTask = new Deadline(taskName, dateTime);
+        tasks.add(newTask);
+
+        save(storage, tasks);
+
+        ui.showReply(String.format("Got it. I've added this task:\n  %s\nNow you have %d tasks in the list.", newTask,
```

Comment:
Our static analysis engine detects that your pull request includes a change that a format string `\n` instead of `%n`. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline), it is a bad practice, since using `\n` in format string may cause cross-platform issues.